runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -216,8 +222,15 @@ public void addStreamToIgnore(String streamName) {
      *
      */
     private void summonNecromancer() {
-        necromancer = Executors.newSingleThreadExecutor(new ThreadFactoryBuilder()
-                .setNameFormat("necromancer-%d").build());
+        // Note that the queue implementation requires the corePoolSize to
+        // be equal to maximumPoolSize, so this should be fine for a single threaded
+        // executor
+        necromancer = new ThreadPoolExecutor(1, 1,
+                0L, TimeUnit.MILLISECONDS,
+                new BoundedQueue<>(numberOfPendingFutures),
+                new ThreadFactoryBuilder()
+                        .setNameFormat("FastObjectLoaderReaderThread-%d").build());
+
         futureList = new ArrayList<>();
```

Comment:
how will the futureList be cleaned up ? That one keeps growing in size with completed futures ?

Reply:
I removed the array, its actually not needed.

Reply:
I updated the patch.